report/renderer/report-utils.js

```diff
@@ -10,6 +10,20 @@ import {Globals} from './report-globals.js';
 const SCREENSHOT_PREFIX = 'data:image/jpeg;base64,';
 const RATINGS = Util.RATINGS;
 
+// Entity sort orders for each audit. This information is static and thus need not be in LHR.
+// In future, this could move to each audit and could be extracted during a build.
+/** @type {Record<string, LH.Audit.Details.TableSortOrder>} */
+const ENTITY_SORT_DIRECTIVES = {
```

Comment:
Honestly I would prefer to have this information in the LHR rather than do this. This strikes me as something that is going to be technical debt moving forward. What if a plugin want's to change ordering?

Another question is why is this necessary in the first place? Shouldn't the ordering of items in the audit details serve as the canonical ordering that the renderer should use?

Reply:
I'm on the fence regarding the first point, and could be convinced to move this to LHR. The tradeoff is additional LHR payload that serves no value to consumers, and is mostly static in nature. That said, headings in LHR too are static, and yet we've included them in LHR. So this is likely an unnecessary savings optimization.

The reason why this directive is necessary though, is because the sort order of individual items could be different than the sort order post grouping by entities. For example, two individual resources might not have made it to the top in our flat list before, but when grouped together, it could end up being right on top. I.e., ordering of items do change at the reporting side post grouping, and the report renderer has no way of identifying the original sorting logic.

Reply:
> and is mostly static in nature

FWIW I don't think we should consider this static if the sort varies between different audits.

> For example, two individual resources might not have made it to the top in our flat list before, but when grouped together, it could end up being right on top.

Sorry not sure I follow. Are you saying the we need the sort order to determine the ordering of entities, even if we could determine the correct order of individual items?

Reply:
> even if we could determine the correct order of individual items

Report side doesn't have this information. The `details.item` array is already sorted and ReportRenderer doesn't have a way to figure out which columns were prioritized for sorting.

To make the example more clear, look at this Unused Javascript audit result, pre-grouping:
![image](https://user-images.githubusercontent.com/683500/215874544-c2f3e66d-1415-4c4b-a3bc-69b87ff976e2.png)

The same audit result post-grouping, would look like this:
![image](https://user-images.githubusercontent.com/683500/215875001-c66b03cb-4ea2-4c8d-b6fd-e9f7a8a7caca.png)

The entity that the earlier report called out (Concert) is no longer at the top. However, to determine the correct sort columns and their order, we'll need that information be passed over from audit to ReportRenderer. Hope that makes sense?

Reply:
I need to find out why the count regressed (15 vs. 16).

Reply:
Distributed this back to audits.